pkg/controllers/machinesetsync/machineset_sync_controller_test.go

```diff
@@ -156,56 +155,280 @@ var _ = Describe("With a running MachineSetSync controller", func() {
+			&capiv1beta1.MachineSet{},
+			&capav1.AWSCluster{},
+			&capav1.AWSMachineTemplate{},
+		)
 	})
 
-	Context("when the MAPI machine set has MachineAuthority set to Cluster API", func() {
+	Context("when the infrastructure resources exist", func() {
 		BeforeEach(func() {
-			By("Creating the CAPI and MAPI MachineSets")
-			mapiMachineSet = mapiMachineSetBuilder.Build()
-			capiMachineSet = capiMachineSetBuilder.WithReplicas(int32(4)).Build()
-
-			Expect(k8sClient.Create(ctx, capiMachineSet)).Should(Succeed())
-			Expect(k8sClient.Create(ctx, mapiMachineSet)).Should(Succeed())
+			Expect(k8sClient.Create(ctx, capaCluster)).To(Succeed(), "capa cluster should be able to be created")
+			Expect(k8sClient.Create(ctx, capaMachineTemplate)).To(Succeed(), "capa machine template should be able to be created")
 
-			By("Setting the AuthoritativeAPI to ClusterAPI")
-			Eventually(k.UpdateStatus(mapiMachineSet, func() {
-				mapiMachineSet.Status.AuthoritativeAPI = machinev1beta1.MachineAuthorityClusterAPI
-			})).Should(Succeed())
 		})
 
-		// For now only happy path
-		It("should update the synchronized condition on the MAPI MachineSet", func() {
-			Eventually(k.Object(mapiMachineSet), timeout).Should(
-				SatisfyAll(
-					HaveField("Status.Conditions", ContainElement(
+		Context("when the MAPI machine set has MachineAuthority set to Cluster API", func() {
+
+			Context("when the spec differs", func() {
+				BeforeEach(func() {
+					By("Creating the CAPI and MAPI MachineSets")
+					mapiMachineSet = mapiMachineSetBuilder.Build()
+					capiMachineSet = capiMachineSetBuilder.WithReplicas(int32(4)).Build()
+
+					Expect(k8sClient.Create(ctx, capiMachineSet)).Should(Succeed())
+					Expect(k8sClient.Create(ctx, mapiMachineSet)).Should(Succeed())
+
+					By("Setting the AuthoritativeAPI to ClusterAPI")
+					Eventually(k.UpdateStatus(mapiMachineSet, func() {
+						mapiMachineSet.Status.AuthoritativeAPI = machinev1beta1.MachineAuthorityClusterAPI
+					})).Should(Succeed())
+				})
+
+				It("should update the synchronized condition on the MAPI MachineSet", func() {
+					Eventually(k.Object(mapiMachineSet), timeout).Should(
 						SatisfyAll(
-							HaveField("Type", Equal(consts.SynchronizedCondition)),
-							HaveField("Status", Equal(corev1.ConditionTrue)),
-						))),
-					HaveField("Status.SynchronizedGeneration", Equal(capiMachineSet.GetGeneration())),
-				))
+							HaveField("Status.Conditions", ContainElement(
+								SatisfyAll(
+									HaveField("Type", Equal(consts.SynchronizedCondition)),
+									HaveField("Status", Equal(corev1.ConditionTrue)),
+								))),
+							HaveField("Status.SynchronizedGeneration", Equal(capiMachineSet.GetGeneration())),
+						))
+				})
+
+				It("should update the replica count", func() {
+					Eventually(k.Object(mapiMachineSet), timeout).Should(
+						HaveField("Spec.Replicas", Equal(ptr.To(int32(4)))),
+					)
+				})
+			})
+
+			Context("when the object meta differs", func() {
```

Comment:
What about when the object meta differs for cases that we don't care about?

Reply:
yep forgot this case, will add a 'when we don't care about the change' context :P